core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1324,9 +1324,9 @@ object KafkaConfig {
       .define(SslEngineFactoryClassProp, CLASS, null, LOW, SslEngineFactoryClassDoc)
 
       /** ********* Sasl Configuration ****************/
-      .define(SaslMechanismInterBrokerProtocolProp, STRING, Defaults.SaslMechanismInterBrokerProtocol, MEDIUM, SaslMechanismInterBrokerProtocolDoc)
+      .define(SaslMechanismInterBrokerProtocolProp, STRING, Defaults.SaslMechanismInterBrokerProtocol, ConfigDef.CompositeValidator.of(new ConfigDef.NonNullValidator(), new ConfigDef.NonEmptyString()), MEDIUM, SaslMechanismInterBrokerProtocolDoc)
       .define(SaslJaasConfigProp, PASSWORD, null, MEDIUM, SaslJaasConfigDoc)
-      .define(SaslEnabledMechanismsProp, LIST, Defaults.SaslEnabledMechanisms, MEDIUM, SaslEnabledMechanismsDoc)
+      .define(SaslEnabledMechanismsProp, LIST, Defaults.SaslEnabledMechanisms, new BrokerSecurityConfigs.SaslEnabledMechanismsValidator(), MEDIUM, SaslEnabledMechanismsDoc)
```

Comment:
I don't think we can add this change; it's possible to bring up a Kafka server right now with `sasl.enabled.mechanisms` set to `,,,` as long as SASL isn't enabled.

Reply:
Why do you need to set `sasl.enabled.mechanisms` to `,,,` if SASL is not enabled. Instead of removing `sasl.enabled.mechanisms` configuration.
In fact, when SASL is not enabled, configuring various mechanisms in `sasl.enabled.mechanisms` will not affect the startup of the broker. I can promise this.
eg:
`sasl.enabled.mechanisms=GSSAPI,PLAIN,SCRAM-SHA-256,SCRAM-SHA-512,OAUTHBEARER`

Reply:
> Is it possible today to bring up a MirrorMaker 2 instance with security.protocol set to SASL without it failing? 


According to the official website https://kafka.apache.org/documentation/#producerconfigs_security.protocol, the valid 
values of `security.protocol` are: PLAINTEXT, SSL, SASL_PLAINTEXT, SASL_SSL.
Currently this legal values apply to Producer, Consumer, KafkaConnect, KafkaStreams, Adminclient.
I'm also not sure if there are different legal values for the `security.protocol` parameter in `Mirrormaker`. If it is indeed different, I think we can do a separate treatment for `Mirrormaker`.
As far as I know, SASL is just a switch that turns on the authentication mechanism, it is not a specific communication protocol. It needs to be used with the transport layer protocol. For example: SASL_PLAINTEXT or SASL_SSL.

WDYT?
Thanks.

Reply:
(@RivenSun2 I think your second comment was meant as a reply to [this one](https://github.com/apache/kafka/pull/12010#discussion_r852477821), right?)

In response to the first comment:

Similar point as above, but it's less about having to permit a value of `,,,` because people would _want_ to set it, and more because it's not worth breaking things for people who already do have it set and can get bitten by the upgrade.

Interesting that this validator doesn't seem to take effect when SASL isn't enabled on the broker, though. Have you tested this change at all? I understand there are unit tests; I'm more wondering if it actually causes a different, more-useful error message to get logged to the user when this property is invalid (and is actually used by the broker)?

Reply:
I think this validator still executes when SASL is not enabled on the broker. Because the validator is executed when KafkaConfig (extends AbstractConfig) is initialized.

I'll verify your case later.

Reply:
On a second thought, SASL modules are highly flexible and customizable.
And this parameter is not a required parameter, which may cause compatibility problems when adding a validator.
So to not make things more complicated, remove the validator for this parameter.
Thanks.